### PR TITLE
Add a showcase devServer

### PR DESCRIPTION
**Pull request by nmattia**
This adds a new webpack configuration and `npm run start` script that
serves a showcase of some of II's pages (right now only the simplest
ones, more to come soon).

This should greatly improve productivity when working on HTML and CSS
(and some JS) changes, since one does not have to go through II flows to
land on a particular page.

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -50,106 +51,175 @@ const devtool = isProduction ? undefined : "source-map";
 /**
  * Generate a webpack configuration for a canister.
  */
-module.exports = {
```
Comment (nmattia) [inline webpack.config.js:53]: it looks like a lot has changed here, but it's basically just formatting. A new config was added below; everything in the old config (except for the `name` field that was added) is unchanged.

File: webpack.config.js [hunk 2]
```diff
@@ -50,106 +51,174 @@ const devtool = isProduction ? undefined : "source-map";
...
-            return path + queryParamsString;
-          },
-        })
-      );
+    devtool,
+    optimization: {
+      minimize: isProduction,
+    },
+    resolve: {
+      extensions: [".js", ".ts", ".jsx", ".tsx"],
```
Comment (peterpeterparker) [inline webpack.config.js:67]: `.jsx` and `.tsx` can be removed I guess here

File: webpack.config.js [hunk 3]
```diff
@@ -50,106 +51,174 @@ const devtool = isProduction ? undefined : "source-map";
...
+      fallback: {
```
Comment (peterpeterparker) [inline webpack.config.js:68]: you need these fallback?

File: webpack.config.js [hunk 4]
```diff
@@ -50,106 +51,174 @@ const devtool = isProduction ? undefined : "source-map";
...
+            // Read the replica host from dfx.json. The replica does not need to be running, which is convenient
+            // in case we just load pages that don't talk to the canister (then we wouldn't want to have to start the replica or build the canister)
+            target: readReplicaHost(),
+
+            // For path rewriting we use a function that lazily (i.e. when first called, using a singleton) reads
+            // the canister ID from the local dfx state and returns a new request path that includes
+            // the ?canisterId=foo atrocity
+            pathRewrite: (pathAndParams, req) => {
+              let queryParamsString = `?`;
+              console.log("One");
```
Comment (peterpeterparker) [inline webpack.config.js:100]: "one"?

File: webpack.config.js [hunk 5]
```diff
@@ -50,106 +51,174 @@ const devtool = isProduction ? undefined : "source-map";
...
+      proxy: {
+        // Make sure /api calls land on the replica (and not on webpack)
+        "/api": "http://localhost:8000",
       },
+      allowedHosts: [".localhost", ".local", ".ngrok.io"],
+    },
+
+    module: {
+      rules: [
+        { test: /\.(ts|tsx)$/, loader: "ts-loader" },
```
Comment (peterpeterparker) [inline webpack.config.js:129]: tsx here too

**Review by peterpeterparker — APPROVED** [review]
LGTM

**Comment by nmattia** [comment]
@peterpeterparker thanks, actually all these were either copied from the old config and/or already there before this PR. I'll submit a clean up PR addressing that.

**Pull request merged**